clients/admin-ui/src/pages/dataset/[datasetId]/index.tsx

```diff
@@ -174,12 +176,14 @@ const DatasetDetailPage: NextPage = () => {
         </Box>
       )}
 
-      <EditCollectionDrawer
-        dataset={dataset!}
-        collection={selectedCollectionForEditing}
-        isOpen={isEditingCollection}
-        onClose={() => setIsEditingCollection(false)}
-      />
+      {dataset && selectedCollectionForEditing && (
```

Comment:
Nice, avoiding needing to use the ! or handling undefined values further down.

Reply:
IMO we should avoid `!` always. It's usually an indication for a potential bug, as was the case here.